osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/InvocationCounterManagerTest.java

```diff
@@ -93,4 +101,41 @@ void initializesInvocationCounterForDevice() throws Exception {
     assertThat(this.device.getVersion()).isGreaterThan(this.initialDeviceVersion);
     assertThat(this.device.getInvocationCounter()).isEqualTo(invocationCounterValueOnDevice);
   }
+
+  @Test
+  void attemptToLowerInvocationCounterForDeviceLogsErrorAndThrowsException() throws Exception {
+    final Logger invocationManagerLogger =
+        (Logger) LoggerFactory.getLogger(InvocationCounterManager.class);
+    final ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+    final List<ILoggingEvent> logsList = listAppender.list;
+    final long invocationCounterValueOnDevice = 1;
+    final DlmsConnectionManager connectionManager = mock(DlmsConnectionManager.class);
+    final DataObject dataObject = DataObject.newUInteger32Data(invocationCounterValueOnDevice);
+
+    when(this.dlmsHelper.getAttributeValue(
+            eq(connectionManager), refEq(ATTRIBUTE_ADDRESS_INVOCATION_COUNTER_VALUE)))
+        .thenReturn(dataObject);
+
+    listAppender.start();
+    invocationManagerLogger.addAppender(listAppender);
+
+    final FunctionalException thrownException =
+        catchThrowableOfType(
+            () ->
+                this.manager.initializeWithInvocationCounterStoredOnDeviceTask(
+                    this.device, connectionManager),
+            FunctionalException.class);
+
+    assertThat(logsList.size() > 0).isTrue();
```

Comment:
Something like: `assertThat(logsList).isNotEmpty();` would give nicer exception messages.

Reply:
right, I have updated this line to the proposed assertThat(logsList).isNotEmpty();